docs/api.yaml

```diff
@@ -0,0 +1,1061 @@
+      responses:
+        200:
+          description: "Collection of created students"
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Student'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+    patch:
+      tags:
+        - "Students"
+      summary: "Update an existing student"
+      parameters:
+        - name: "id"
+          in: "query"
+          description: "Identifier of the student to update"
+          required: true
+          schema:
+            type: string
+      description: |
+        Possible udpates on:
+        * Group: change
+        * Status: change
+        * EnrolledCourses: add
+        * Grades : add
+      operationId: "updateStudent"
+      requestBody:
+        description: "Student to update"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Student'
+            examples:
+              addGroup:
+                summary: Change group
+                #  description: ""
+                value:
+                  group:
+                    id: "uuid"
+              changeStatus:
+                summary: Change status
+                #description: ""
+                value:
+                  status: "ENABLED"
+              addEnrolledCourse:
+                summary: Add enrolled course
+                value:
+                  course_id: "uuid"
+                  course_name: "Architecture logicielle"
+              addGrade:
+                summary: Add grade to an enrolled course
+                value:
+                  value: "0"
+                  enrolled_course_id: "uuid"
+                  enrolled_course_name: "Architecture logicielle"
+
+      responses:
+        200:
+          description: "The updated Student"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Student'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+    delete:
+      tags:
+        - "Students"
+      summary: "Delete a student"
+      operationId: "deleteStudent"
+      parameters:
+        - name: "id"
+          in: "query"
+          description : "Identifier of the student that needs to be deleted"
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: "Student deleted"
```

Comment:
No. Be more generic to allow re-usability. Create a new 200Delete response described as "Resources deleted" (mind the plural)